src/App.tsx

```diff
@@ -1,24 +1,287 @@
+    try {
+      const todosFromServer = await getTodos(USER_ID);
+
+      setTodos(todosFromServer);
+    } catch {
+      setErrorMessage('Unable to fetch a todo');
+      setTodos([...todos]);
+    }
+  };
+
+  const addTodo = async (title: string) => {
+    setTempTodo((state) => ({
+      ...state,
+      title,
+    }));
+    setAdded(true);
+    setSearchQuery('');
+    try {
+      const addedResultFromServer = await addTodos(USER_ID, {
+        title,
+        userId: USER_ID,
+        completed: false,
+      });
+
+      if (addedResultFromServer) {
+        fetchTodos();
+        setTempTodo((state) => ({
+          ...state,
+          title: '',
+        }));
+        setAdded(false);
+      }
+    } catch {
+      setErrorMessage('Unable to add a todo');
+      setAdded(false);
+      setTodos([...todos]);
+    }
+  };
+
+  const updateTodo = async (
+    todoId: number,
+    status: boolean | string,
+    by: string,
+  ) => {
+    setEditingId(0);
+    setUpdatedId(todoId);
+
+    const updateData = typeof status === 'boolean'
+      ? { completed: !status }
+      : { title: status };
+
+    try {
+      const addedResultFromServer = await updateTodos(todoId, updateData);
+
+      if (addedResultFromServer) {
+        setTodos((state) => {
+          return state.map(todo => {
+            if (by !== 'title' && todo.id === todoId) {
+              return {
+                ...todo,
+                completed: !todo.completed,
+              };
+            }
+
+            return todo;
+          });
+        });
+        fetchTodos();
+      }
+    } catch (e) {
+      setErrorMessage('Unable to update a todo');
+    }
+
+    setUpdatedId(0);
+    setIsLoading(false);
+  };
+
+  const onEmpty = () => {
+    setErrorMessage('Title can\'t be empty');
+  };
+
+  const todoDelete = async (todoId: number) => {
+    setDeletedId(todoId);
+    try {
+      const deleteResultFromServer = await deleteTodos(todoId);
+
+      if (deleteResultFromServer) {
+        setTodos((state) => state.filter(todo => todo.id !== todoId));
+        fetchTodos();
+      }
+    } catch {
+      setErrorMessage('Unable to delete a todo');
+    }
+
+    setDeletedId(0);
+    setEditingId(0);
+    setIsLoading(false);
+  };
+
+  const completeAllToggle = async () => {
+    setUpdatedId('allsame');
+    const allToggle = todos.filter(todo => !todo.completed).length !== 0;
+
+    try {
+      const changedElements = await Promise.all(
+        todos.map(todo => updateTodos(todo.id, { completed: allToggle })),
+      );
+
+      if (changedElements) {
+        setTodos((state) => {
+          return state.map(todo => {
+            return {
+              ...todo,
+              completed: allToggle,
+            };
+          });
+        });
+        fetchTodos();
+      }
+    } catch {
+      setErrorMessage('Unable to update a todo');
+    }
+
+    setUpdatedId(0);
+  };
+
+  const clearAllCompleted = async () => {
+    setUpdatedId('all');
```

Comment:
How the id can be `all`? Use an enum and rename the variable if you really need it